src/otx/algo/object_detection_3d/heads/depth_predictor.py

```diff
@@ -0,0 +1,151 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""depth predictor transformer head for 3d object detection."""
+
+from __future__ import annotations
+
+from typing import Callable
+
+import torch
+from torch import nn
+from torch.nn import functional
+
+from otx.algo.common.layers.transformer_layers import TransformerEncoder, TransformerEncoderLayer
+
+
+class DepthPredictor(nn.Module):
+    """Depth predictor and depth encoder."""
+
+    def __init__(
+        self,
+        depth_num_bins: int,
+        depth_min: float,
+        depth_max: float,
+        hidden_dim: int,
+        activation: Callable[..., nn.Module] = nn.ReLU,
```

Comment:
Could you add docstring for this?

Reply:
Done https://github.com/openvinotoolkit/training_extensions/pull/4031/commits/00200f4d4811d82e86e371cb5dabd2541098e83c